### PR TITLE
AK-25042  - Add segment options support for Fortinet

### PR DESCRIPTION
**Pull request by mac-alkira**

### CODE REVIEW

File: alkira/resource_alkira_service_fortinet_helper.go [hunk 1]
```diff
@@ -32,3 +34,75 @@ func expandFortinetInstances(in *schema.Set) []alkira.FortinetInstance {
 
 	return instances
 }
+
+func expandFortinetSegmentOptions(in *schema.Set, m interface{}) (map[string]alkira.FortinetSegmentName, error) {
+	if in == nil || in.Len() == 0 {
+		return nil, errors.New("Fortinet segment options cannot be null or empty")
```
Comment (spikewang) [inline alkira/resource_alkira_service_fortinet_helper.go:40]: The segment options should be optional?

File: alkira/resource_alkira_service_fortinet.go [hunk 2]
```diff
@@ -112,6 +112,31 @@ func resourceAlkiraServiceFortinet() *schema.Resource {
 				Required:    true,
 				Elem:        &schema.Schema{Type: schema.TypeString},
 			},
+			"segment_options": {
+				Type:        schema.TypeSet,
+				Required:    true,
```
[the remark this answers is not in the document]
Comment (mac-alkira) [reply alkira/resource_alkira_service_fortinet.go:117]: It is listed as optional in the external api docs, but it is required. I tried to omit it and it failed so I added it as required here to help the user avoid the issue.

File: alkira/resource_alkira_service_fortinet.go [hunk 3]
```diff
@@ -112,6 +112,31 @@ func resourceAlkiraServiceFortinet() *schema.Resource {
...
+				Description: "The segment options as used by your checkpoint firewall.",
```
[the remark this answers is not in the document]
Comment (mac-alkira) [reply alkira/resource_alkira_service_fortinet.go:118]: Nice catch. Thanks.

File: alkira/resource_alkira_service_fortinet.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (mac-alkira) [reply alkira/resource_alkira_service_fortinet.go:117]: I'll switch it back if you want. It just means the user will have to deal with the API error directly.

File: alkira/resource_alkira_service_fortinet_helper.go (hunk 1 above)
Comment (mac-alkira) [reply alkira/resource_alkira_service_fortinet_helper.go:40]: see above.

**Comment by spikewang** [comment]
@mac-alkira Don't forget to regenerate the doc...

**Review by spikewang — APPROVED** [review]

**Pull request merged**